src/js/components/TextInput/TextInput.js

```diff
@@ -72,6 +72,7 @@ class TextInput extends Component {
   state = {
     activeSuggestionIndex: -1,
     showDrop: false,
+    isInputEmpty: true,
```

Comment:
What if an initial value is specified? It seems like the placeholder will be shown in that case?

If that's correct, I'm wondering if we don't need the check for value length in `onChange` or a state, but just a change to the value test in the `render` function, such as?:
`placeholder && typeof placeholder !== 'string' && ((typeof value === 'string' && value.length === 0) || value === undefined)`

Reply:
nice point.
i didn't test for a controlled component.
in this case i not showing the placeholder is the expected behavior.